app/models/service_template_ansible_playbook.rb

```diff
@@ -117,4 +109,25 @@ def self.validate_config_info(info)
     info
   end
   private_class_method :validate_config_info
+
+  def job_template(action)
+    resource_actions.find_by!(:action => action.to_s.capitalize).configuration_template
+  end
+
+  def update_catalog_item(options, auth_user = nil)
+    config_info = validate_update_config_info(options)
+    name = options[:name]
+    description = options[:description]
+    [:provision, :retirement, :reconfigure].each do |action|
+      next unless config_info[action]
+      info = config_info[action]
+      self.class.send(:create_new_dialog, info[:new_dialog_name], job_template(action)) if info[:new_dialog_name]
```

Comment:
`create_new_dialog` is now instance method that you can directly call.
Also just like `create_catalog_item`, we need to update `options[:config_info][action][:dialog_id]` to the newly created dialog id.